lsp/nls/src/requests/completion.rs

```diff
@@ -167,30 +173,61 @@ fn record_path_completion(term: RichTerm, world: &World) -> Vec<CompletionItem>
 }
 
 // Try to complete a field name in a record, like in
+
 // ```
 // { bar = 1, foo }
 //               ^cursor
 // ```
+
 // In this situation we don't care about the environment, but we do care about
 // contracts and merged records.
-fn field_completion(rt: &RichTerm, world: &World) -> Vec<CompletionItem> {
+//
+// If `path` is non-empty, instead of completing fields of `rt` we complete
+// the fields of `rt.<path>`. You might think we'd want to do this in a situation like
+
+// ```
+// { bar = 1, foo.blah.ba }
+//                       ^cursor
+// ```
+
+// but in fact the nickel parser has already expanded this to
+
+// ```
+// { bar = 1, foo = { blah = { ba } } }
+// ```
+//
+// so we don't encounter the path in this case. Instead, the non-empty path only comes
+// into play when the input fails to parse completely, like in (note the trailing dot)
+
+// ```
+// { bar = 1, foo.blah.ba. }
+//                        ^cursor
+// ```
```

Comment:
Nitpick: we can use docstrings for private item (`///`), and I think the accepted style when using an empty line is to still prefix it with `///` (otherwise it's not clear that we're still documenting the same item)